doc/SAI_1.6.5_ReleaseNotes.md

```diff
@@ -0,0 +1,539 @@
+
+```
+### saiqosmap.h	
+QOS Map to set traffic class and color to EXP.
+
+```
+ /** QOS Map to set EXP to Traffic class */
+    SAI_QOS_MAP_TYPE_MPLS_EXP_TO_TC = 0x0000000a,
+
+    /** QOS Map to set EXP to color */
+    SAI_QOS_MAP_TYPE_MPLS_EXP_TO_COLOR = 0x0000000b,
+
+    /** QOS Map to set traffic class and color to EXP */
+    SAI_QOS_MAP_TYPE_TC_AND_COLOR_TO_MPLS_EXP = 0x0000000c,
+```
+
+### saiqueue.h	
+H/w Egress Unicast Queue and H/w Multicast Egress (Broadcast, Unknown unicast, Multicast) Queue has been added.
+
+```
+/** H/w Virtual Output Queue (VOQ). This queue is ingress unicast queue */
+    SAI_QUEUE_TYPE_UNICAST_VOQ = 0x00000003,
+
+    /** H/w Virtual Output Queue (VOQ). This queue is fabric multicast queue */
+    SAI_QUEUE_TYPE_MULTICAST_VOQ = 0x00000004,
+
+    /** H/w Fabric Queue. */
+    SAI_QUEUE_TYPE_FABRIC_TX = 0x00000005,
+
+```
+
+### sairouterinterface.h	
+
+In this release, the sai router interface type for system port has been modified as mentioned below
+
+
+```
+   SAI_ROUTER_INTERFACE_ATTR_TYPE
+   
+    @condition SAI_ROUTER_INTERFACE_ATTR_TYPE == SAI_ROUTER_INTERFACE_TYPE_PORT or SAI_ROUTER_INTERFACE_ATTR_TYPE == SAI_ROUTER_INTERFACE_TYPE_SUB_PORT
+	
+```
+
+### saiswitch.h	
+This is a top-level object exposed from the Adapter. This switch object controls the switch level behavior such as cut-through or store-forward, ecmp hash configuration.
+In this release, the below attributes are added in API along with enhancements. 
+
+1)Attribute data for #SAI_SWITCH_ATTR_HARDWARE_ACCESS_BUS, SAI_SWITCH_ATTR_FIRMWARE_LOAD_METHOD, SAI_SWITCH_ATTR_FIRMWARE_LOAD_TYPE, SAI_SWITCH_ATTR_TYPE has been added. <br>
+2)Switch hardware access bus MDIO/I2C/CPLD has been added. <br>
+3)Platform context information provided by the host adapter to driver. This information is Host adapter specific, typically used for maintain synchronization and device information. <br>
+4)Driver will give this context back to adapter as part of call back sai_switch_register_read/write_fn API. <br>
+5)Platform adaption device write callback function passed to the adapter. This is mandatory function for driver when device access not supported by file system. <br>
+6)Platform specific device register read access. This API provides platform adaption functionality to access device registers from driver. This is mandatory to pass as attribute to sai_create_switch when driver implementation does not support register access by device file system directly.<br>
+7)Platform specific device register write access. This API provides platform adaption functionality to access device registers from driver. This is mandatory to pass as attribute to sai_create_switch when driver implementation does not support register access by device file system directly.<br>
+8)Switch MDIO read API - Provides read access API for devices connected to MDIO from NPU SAI.<br>
+9)Switch MDIO write API - Provides write access API for devices connected to MDIO from NPU SAI.<br>
+

```

Comment:
Can you add an item about the new system port related attributes?

SAI_SWITCH_ATTR_SWITCH_ID
SAI_SWITCH_ATTR_MAX_SYSTEM_CORES
SAI_SWITCH_ATTR_SYSTEM_PORT_CONFIG_LIST
SAI_SWITCH_ATTR_NUMBER_OF_SYSTEM_PORTS
SAI_SWITCH_ATTR_SYSTEM_PORT_LIST
SAI_SWITCH_ATTR_NUMBER_OF_FABRIC_PORTS
SAI_SWITCH_ATTR_FABRIC_PORT_LIST